src/NuGet.Services.EndToEnd/Support/Clients/GalleryClient.cs

```diff
@@ -156,6 +157,23 @@ public async Task UnlistAsync(string id, string version, ITestOutputHelper logge
             await SendToPackageAsync(HttpMethod.Delete, id, version, logger);
         }
 
+        public async Task SearchPackageODataV2FromDBAsync(string id, string normalizedVersion, ITestOutputHelper logger)
+        {
+            var galleryEndpoint = await GetGalleryUrlAsync(logger);
+            var url = $"{galleryEndpoint}/api/v2/Packages()?$filter=tolower(Id) eq '{id.ToLower()}' and NormalizedVersion eq '{normalizedVersion}' and 1 eq 1";
```

Comment:
you don't need the `tolower`. `Id eq '{id}'` is enough.

Query strings should be built with proper encoding. See `AppendAutocompletePackageIdsQueryString` below and the use of `QueryHelpers.AddQueryString`

Reply:
Could you add a comment about the `1 eq 1` trick? I didn't realize what this was doing until I found Joel's comment

Reply:
Updated for both comments.